javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js

```diff
@@ -6,15 +6,60 @@ import { reformatJSON } from 'main/utils/BasicCourseTableHelpers';
 import { yyyyqToQyy } from 'main/utils/quarterUtilities';
 import ToolkitProvider from 'react-bootstrap-table2-toolkit';
 
-const BasicCourseTable = ( {classes, checks, displayQuarter, allowExport} ) => {
+
+  const BasicCourseTable = ( {classes, checks, displayQuarter, allowExport} ) => {
   const { isAuthenticated } = useAuth0();
   const sections = reformatJSON(classes,checks);
 
-  
-  const rowStyle = (row, _rowIndex) => {
-    return  (row.section % 100 === 0)? {backgroundColor: '#CEDEFA'}: {backgroundColor: '#EDF3FE'};
-  }
+  const COLOR_UNAVAILABLE = {backgroundColor: '#FF0000'};
+  const COLOR_CLOSEFULL = {backgroundColor: '#FFBF00'};
+  const COLOR_AVAILABLELECTUREORCLASSWITHSECTIONS = {backgroundColor: '#CEDEFA'};
+  const COLOR_AVAILABLESECTION = {backgroundColor: '#EDF3FE'};
+  const CLOSEFULL_THRESHOLD=0.2;
+  const classUnavailable = (row) => (row.enrolledTotal >= row.maxEnroll || row.courseCancelled === "Y" || row.classClosed ==="Y"); 
+  const closeToFull = (row) => ((row.maxEnroll - row.enrolledTotal) < (CLOSEFULL_THRESHOLD * row.maxEnroll));
 
+  const rowStyle = (row) => {
+    if (row.section % 100 === 0)
+    {
+      //We interate through all the classes, for the first section (should be the mod 100 == 0 section) if it is equal to the section we are setting
+      //the color to we do something.
+      var i; 
+      for (i in classes) 
```

Comment:
The syntax here is a little weird, it would make more sense to also rename i to something more user friendly
for (var class in classes ) {}

Or classes.forEach(class=> {}